### PR TITLE
Refactor move servicebus message state key to azure-core-amqp

### PR DESCRIPTION
**Pull request by ZejiaJiang**
# Description
Refactor: move servicebus message state key to `azure-core-amqp` constants. [#26898](https://github.com/Azure/azure-sdk-for-java/issues/26898)

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
you might need to wait after servicebus June release.

**Comment by weidongxu-microsoft** [comment]
I think servicebus is released, you might want to refresh main and get it merged before next code freeze.

**Review by JimSuplizio — COMMENTED** [review]
The creation and use of the unreleased_ tag was done correctly and I'm okay with those changes. I'll let the other people review and approve the corresponding code changes.

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/AmqpMessageConstant.java [hunk 1]
```diff
@@ -122,7 +122,12 @@ public enum AmqpMessageConstant {
     /**
      * The identifier for deadletter reason.
      */
-    DEAD_LETTER_REASON_ANNOTATION_NAME("DeadLetterReason");
+    DEAD_LETTER_REASON_ANNOTATION_NAME("DeadLetterReason"),
+    /**
+     * The state of message.
+     */
+    SERVICE_BUS_MESSAGE_STATE_ANNOTATION_NAME("x-opt-message-state");
```
[the remark this answers is not in the document]
Comment (ZejiaJiang) [reply sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/AmqpMessageConstant.java:129]: Yeah, the above keys have `ANNOTATION_NAME` suffix as well and we do get the key from annotation.

I'll check these keys usage and remove the prefix.

**Review by conniey — APPROVED** [review]

**Review by anuchandy — APPROVED** [review]

**Pull request merged**